include/deal.II/lac/vector.templates.h

```diff
@@ -48,6 +48,23 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+/**
+ * Helper struct to tell us if we can use SIMD instructions for the given @p
+ * Number type.
+ */
+template <typename Number>
+struct UseSimdFor
+{
+  static const bool value = true;
+};
+
+// disable long double SIMD
+template <>
+struct UseSimdFor<long double>
+{
+  static const bool value = false;
+};
```

Comment:
That is true *as of now* and *for x86_64*. But it doesn't have to be true in the future or for other systems.